libkbfs/folder_block_ops.go

```diff
@@ -1256,45 +1310,131 @@ func (fbo *folderBlockOps) Write(
 	return nil
 }
 
-// Returns the set of newly-ID'd blocks created during this truncate
-// that might need to be cleaned up if the truncate is deferred.
-func (fbo *folderBlockOps) truncateLocked(
+// truncateExtendLocked is called by truncateLocked to extend a file and
+// creates a hole.
+func (fbo *folderBlockOps) truncateExtendLocked(
 	ctx context.Context, lState *lockState, md *RootMetadata,
-	file path, size uint64) (*WriteRange, []BlockPointer, error) {
-	fbo.blockLock.AssertLocked(lState)
+	file path, size uint64) (WriteRange, []BlockPointer, error) {
 
-	// check writer status explicitly
-	username, uid, err := fbo.config.KBPKI().GetCurrentUserInfo(ctx)
+	if size > fbo.config.MaxFileBytes() {
+		return WriteRange{}, nil, FileTooBigError{file, int64(size), fbo.config.MaxFileBytes()}
+	}
+
+	fblock, uid, err := fbo.writeGetFileLocked(ctx, lState, md, file)
 	if err != nil {
-		return nil, nil, err
+		return WriteRange{}, nil, err
 	}
-	if !md.GetTlfHandle().IsWriter(uid) {
-		return nil, nil, NewWriteAccessError(md.GetTlfHandle(), username)
+
+	var dirtyPtrs []BlockPointer
+
+	fbo.log.CDebugf(ctx, "truncateExtendLocked: extending fblock %#v", fblock)
+	if !fblock.IsInd {
```

Comment:
Sorry, one last thing -- if `fblock.IsInd` is already true, what sets the data version to 2?